src/test-run/index.ts

```diff
@@ -1233,10 +1233,10 @@ export default class TestRun extends AsyncEventEmitter {
     }
 
     public async _makeScreenshotOnFail (failedActionId?: string): Promise<void> {
-        const { screenshots } = this.opts;
+        const { screenshots } = this.opts as { screenshots: ScreenshotOptionValue };
 
-        if (!this.errScreenshotPath && (screenshots as ScreenshotOptionValue)?.takeOnFails)
-            this.errScreenshotPath = await this._internalExecuteCommand(new browserManipulationCommands.TakeScreenshotOnFailCommand({ failedActionId })) as string;
+        if (!this.errScreenshotPath && screenshots?.takeOnFails)
+            this.errScreenshotPath = await this._internalExecuteCommand(new browserManipulationCommands.TakeScreenshotOnFailCommand({ failedActionId, fullPage: screenshots?.fullPage })) as string;
```

Comment:
```suggestion
            this.errScreenshotPath = await this._internalExecuteCommand(new browserManipulationCommands.TakeScreenshotOnFailCommand({ failedActionId, fullPage: screenshots.fullPage })) as string;
```
It looks like there is no need to check if `screenshot` variable is not empty twice